lib/spack/spack/spec.py

```diff
@@ -4457,7 +4445,7 @@ def __hash__(self):
         # This is good for large, unchanging specs.
         if self.concrete:
             if not self._dunder_hash:
-                self._dunder_hash = self.dag_hash_bit_prefix(64)
+                self._dunder_hash = self.process_hash_bit_prefix(64)
```

Comment:
This needs a comment about possible recursion, as well.

Reply:
I agree with your https://github.com/spack/spack/pull/28504/files#r826763169 :slightly_smiling_face: